src/order/succ_pred/basic.lean

```diff
@@ -753,6 +753,11 @@ section preorder
 variables [preorder α]
 
 section succ_order
+
+lemma is_succ_archimedean.of_dual [succ_order α] [is_pred_archimedean (order_dual α)] :
+  is_succ_archimedean α :=
+⟨λ a b h, by convert exists_pred_iterate_of_le h.dual⟩
```

Comment:
Why this? You can use `order_dual.is_succ_archimedean`. Yes the types aren't correct, but we use this trick everywhere in the order library and all this dualization kerfuffle will go away once we have the long sought `to_dual` attribute.

Reply:
Use `order_dual.is_succ_archimedean` instead of `is_succ_archimedean.of_dual` at the place of use.
What should work on this line is
```suggestion
lemma is_succ_archimedean.of_dual [succ_order α] [is_pred_archimedean (order_dual α)] :
  is_succ_archimedean α :=
@order_dual.is_succ_archimedean (order_dual (order_dual α) _ _ _
```

Reply:
(that doesn't seem to work either)